map/src/map/components/ContextMenu.jsx

```diff
@@ -119,16 +115,181 @@ export default function ContextMenu({ setGeocodingData, setRegionData }) {
         });
     }
 
-    function addFavorite(e) {
-        ctx.addFavorite.location = e.latlng;
+    function addFavorite(latlng) {
+        ctx.addFavorite.location = latlng;
         ctx.setAddFavorite({ ...ctx.addFavorite });
     }
 
     function formattingRegions(regions) {
         return regions.join(', ').replaceAll('_', ' ');
     }
 
-    function copyCoordinates(e) {
-        navigator.clipboard.writeText(e.latlng.lat.toFixed(6) + ',' + e.latlng.lng.toFixed(6));
+    function copyCoordinates(latlng) {
+        navigator.clipboard.writeText(latlng.lat.toFixed(6) + ',' + latlng.lng.toFixed(6));
     }
+
+    function openWeather(latlng) {
+        ctx.setOpenMenu({ id: 'se-show-menu-weather', latlng: latlng });
+    }
+
+    return (
+        <>
+            <div
+                ref={anchorEl}
+                style={{
+                    position: 'absolute',
+                    top: menuPosition?.mouseY || 0,
+                    left: menuPosition?.mouseX || 0,
+                    width: 1,
+                    height: 1,
+                    pointerEvents: 'none',
+                }}
+            />
+            <Popover
+                anchorOrigin={{
+                    vertical: 'top',
+                    horizontal: 'left',
+                }}
+                transformOrigin={{
+                    vertical: 'top',
+                    horizontal: 'left',
+                }}
+                open={open}
+                anchorEl={anchorEl.current}
+                disablePortal={true}
+            >
+                <ClickAwayListener
```

Comment:
1) broken closing menu by Escape button (was OK with the previous menu)

2) closing by click-away on main menu should not add a new point in Track editor mode (that's old bug)